python/pyarrow/dataset.py

```diff
@@ -798,6 +800,20 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         used determined by the number of available CPU cores.
     max_partitions : int, default 1024
         Maximum number of partitions any batch may be written into.
+    max_open_files : int, default 1024
+        Maximum number of number of files can be opened
+    max_rows_per_file : int, default 0
+        Maximum number of rows per file
```

Comment:
Same here, helpful to have the extra guidance:

```suggestion
        Maximum number of rows per file. If greater than 0 then this will 
        limit how many rows are placed in any single file. Otherwise there 
        will be no limit and one file will be created in each output directory 
        unless files need to be closed to respect max_open_files
```

Reply:
Added the full C++ docstrings.